### PR TITLE
feat(useTransition): support for vectors

### PR DESCRIPTION
**Pull request by scottbedard**
This is the first of several PRs to implement features being designed in #364

There is a small breaking change here. The returned value is now a `ComputedRef` instead of a `Ref`, meaning anyone who manually set the value of a transitioned ref would no longer be able to do so. This was never part of the documented API though, so I'm not sure how much to worry about it. I think just making a mention in the release notes would be sufficient.

Vectors support any number of dimensions, and there is a quick track/sled example added to the demo.

### CODE REVIEW

**Comment by scottbedard** [comment]
Hmm, I see the deep watching test is not passing in 2.x. It's late for me, so I'll look in the morning and update the PR once it's resolved.

File: packages/shared/utils/is.ts [hunk 1]
```diff
@@ -16,4 +16,8 @@ export const now = () => Date.now()
 export const timestamp = () => +Date.now()
 export const clamp = (n: number, min: number, max: number) => Math.min(max, Math.max(min, n))
 export const noop = () => {}
-export const rand = (min: number, max: number) => Math.floor(Math.random() * (max - min + 1)) + min
+export const rand = (min: number, max: number) => {
+  min = Math.ceil(min)
+  max = Math.floor(max)
+  return Math.floor(Math.random() * (max - min + 1)) + min
+}
```
Comment (scottbedard) [inline packages/shared/utils/is.ts:23]: Reverted, will generate random int between min (inclusive) and max (inclusive)

**Comment by antfu** [comment]
Looks good, do you want to have this in 4.x or 5.0? Both works for me. If you are targeting 5.0, I have just created a new branch `next` for that and you can check the base of this PR. Thanks.

**Comment by scottbedard** [comment]
@antfu Great! Since this does contain that breaking change, we should probably target 5.x. I'll also be submitting a few related PRs soon too, so it'll probably be easiest to just release all the changes together 🤷‍♂️

**Pull request merged**